kernel/src/transaction.rs

```diff
@@ -56,14 +81,14 @@ impl Transaction {
             read_snapshot: snapshot.into(),
             operation: None,
             commit_info: None,
+            write_metadata: vec![],
```

Comment:
note we make `write_metadata` a vec so that we can keep all the batches to traverse multiple times in the case we need to retry the commit. one alternative here is to just require the engine to do the concatenation and hand us one big batch of write metadata

Reply:
Java kernel requires engine to hand over an iterator, potentially multiple times if there are retries. That way it's engine's job to deal with memory management and spilling on big commits.

Reply:
yea I think we could have them pass something that's `impl IntoIterator`?

Reply:
opened #465 to have some more discussion and take this as a follow-up :)